Apps/Src/Voltage.c

```diff
@@ -161,11 +174,12 @@ uint32_t Voltage_GetOpenWire(void){
  * @return voltage of module at specified index
  */
 uint16_t Voltage_GetModuleMillivoltage(uint8_t moduleIdx){
-    // These if statements prevents a hardfault.
+	CPU_TS ts;
+	OSMutexPend(Voltage_Mutex, 0, OS_OPT_PEND_BLOCKING, &ts, &err);
```

Comment:
Can you move this to right before you access the `VoltageVal` array? If you pend the mutex here, there is  chance that you return from this function without posting the mutex.